### DISCUSSION TITLE
Waveshare compatibility

### DISCUSSION THREAD

**Discussion by maskedfighter** [kellerza/sunsynk#364] [General]
Hi!
I wanted to ask wether it is possible to use a waveshare rs485 poe eth device together with this integration? Waveshare supports modbus tcp, so my guess is that i would need the mbusd server to convert it to rtu?

thx for helping! :)

**Comment by kellerza** [comment]
No idea.

You can try these options here if you have a unit https://kellerza.github.io/sunsynk/reference/multi-options#port 

There were also some past discussions on other WaveShare devices - see #96

**Comment by maskedfighter** [comment]
according to waveshare´s website: https://www.waveshare.com/wiki/RS232/485/422_TO_POE_ETH_(B) it should be working (modbus rtu to tcp, tcp output)... i will try it, device arrives on friday, will let u know :)

**Reply by Nurgus** [reply]
Any luck with this? I have this one:

https://www.waveshare.com/wiki/RS485_TO_POE_ETH_(B)

And I'm having no joy at all with it.

**Comment by Nurgus** [comment]
Is there a recommended POE compatible device?

**Comment by proggaras** [comment]
Chck out my PR => https://github.com/kellerza/sunsynk/pull/426/commits/0a2d832a4ce38f25d1b01145a71adb8965b12bfd
I updated the doc exactly with this device. As soon as @kellerza has merged it you find it in the offical manual.

**Reply by Nurgus** [reply]
Huh. I'll give it another shot. My settings look exactly the same as yours. I guess the only thing left for me to look at is cabling.

**Reply by Nurgus** [reply]
![20250210_171034](https://github.com/user-attachments/assets/333c7913-d2dc-4ebf-9e98-d5f753f224a9)
I'm plugged into the ModeBUS in this picture. Have also tried Meter-485. Does this look right? The terminology on my inverter (Sunsynk 8kw) is a bit different to the ones in the documentation.

**Reply by Waz-Cpt** [reply]
You need to use the BMS port on the 8kW unit. If I remember correctly there is a seperate CAN port. I will take pictures tonight for you of my system. My batteries are connected via CAN now. My previous batteries could only connect on the BMS port and then I had to use the serial port for the dongle to connect to this integration.

**Reply by Nurgus** [reply]
Oh wow. What batteries do you have? Mine are the Sunsynk 5kw wall mounted units. I can't see a CAN port..presumably I'm looking in the wrong place.

Am I safe to just unplug the batteries and try them in CAN when I find it? Or do I need to shut things down?

**Reply by Waz-Cpt** [reply]
I have not had a chance to open up my unit to take pictures.

What battery protocol setting are you using for your batteries?

**Reply by Nurgus** [reply]
Protocol is set to 0. I'm thinking now that I need a cable splitter or to use the db9 seriel port instead of the wifi dongle.

**Reply by Nurgus** [reply]
All solved now, I made a splitter and got it working from the BMS port.
